kernel/kernel-api/src/test/scala/io/delta/kernel/internal/PaginationContextSuite.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal
+
+import java.util.Optional
+
+import io.delta.kernel.internal.replay.PaginationContext
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class PaginationContextSuite extends AnyFunSuite {
+
+  test("forFirstPage should create context with empty optionals and specified page size") {
+    val pageSize = 100L
+    val context = PaginationContext.forFirstPage(pageSize)
+
+    assert(!context.getLastReadLogFileName().isPresent)
+    assert(!context.getLastReturnedRowIndex().isPresent)
+    assert(!context.getLastReadSidecarFileIdx().isPresent)
+    assert(context.getPageSize() === pageSize)
+  }
+
+  test("forPageWithPageToken should create context with provided values") {
+    val lastReadLogFileName = "00000000000000000000.json"
+    val lastReturnedRowIndex = 42L
+    val lastReadSidecarFileIdx = Optional.of(java.lang.Long.valueOf(5L))
+    val pageSize = 50L
+
+    val context = PaginationContext.forPageWithPageToken(
+      lastReadLogFileName,
+      lastReturnedRowIndex,
+      lastReadSidecarFileIdx,
+      pageSize)
+
+    assert(context.getLastReadLogFileName().isPresent)
+    assert(context.getLastReadLogFileName().get() === lastReadLogFileName)
```

Comment:
can we just compare context.getLastReadLogFileName() with Optional.of(lastReadLogFileName)?